### PR TITLE
Screenshot Preview with multiple iframes (Closing for future)

### PR DESCRIPTION
**Pull request by mujabmuneeb**
Added a new file (preview.js). There is a new function called generatePreview which receives url, ext, type, width, and height as params and returns blob object.

### CODE REVIEW

File: screenshot-iframe.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    if(hash) {
+        var tempUrl = `https://app.webaverse.com/${hash}.${ext}/preview.${type}`
+    }
+    else {
+        var tempUrl = `https://app.webaverse.com/screenshot.html?url=${reqUrl}&hash=${hash}&ext=${ext}&type=${type}&width=${width}&height=${height}`;
+    }
+
+    const u = url.parse(tempUrl, true);
+
+    const spec = (() => {
```
Comment (abeersaqib) [inline screenshot-iframe.js:39]: what is the use of the function here ?

Can't we construct the url directly ?

File: screenshot-iframe.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if(!iframeExists.length) {
+        var iframe = document.createElement('iframe');
+        iframe.width = '0px';
+        iframe.height = '0px';
+        iframe.name = 'screenshotIframe';
+        document.body.appendChild(iframe);
+    }
+
+    if(hash) {
+        var tempUrl = `https://app.webaverse.com/${hash}.${ext}/preview.${type}`
```
Comment (abeersaqib) [inline screenshot-iframe.js:31]: why are you even constructing it this way ?

File: screenshot-iframe.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        spec = null;
+    })();
+
+    if (spec) {
+        const {url, hash, ext, type, height, width} = spec;
+        var ssUrl = `https://app.webaverse.com/screenshot.html?url=${url}&ext=${ext}&type=${type}&width=${width}&height=${height}`;
+    }
+
+    window.open(ssUrl, 'screenshotIframe')
```
Comment (abeersaqib) [inline screenshot-iframe.js:66]: Have you tested it that it's not re-creating the iframe I don't think so it's tested properly?

**Review by abeersaqib — CHANGES_REQUESTED** [review]

File: screenshot-iframe.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import {storageHost} from './constants';
+import url from 'url';
+
+function parseQuery(queryString) {
```
[the remark this answers is not in the document]
Comment (mujabmuneeb) [reply screenshot-iframe.js]: Removed.

File: screenshot-iframe.js (hunk 1 above)
Comment (mujabmuneeb) [reply screenshot-iframe.js]: Yes we can, fixed it now.

File: screenshot-iframe.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        iframe.height = '0px';
+        iframe.name = 'screenshotIframe';
+        document.body.appendChild(iframe);
+    }
+
+    if(hash) {
+        var tempUrl = `https://app.webaverse.com/${hash}.${ext}/preview.${type}`
+    }
+    else {
+        var tempUrl = `https://app.webaverse.com/screenshot.html?url=${reqUrl}&hash=${hash}&ext=${ext}&type=${type}&width=${width}&height=${height}`;
```
[the remark this answers is not in the document]
Comment (mujabmuneeb) [reply screenshot-iframe.js]: Updated it.

File: screenshot-iframe.js (hunk 3 above)
Comment (mujabmuneeb) [reply screenshot-iframe.js:20]: Yes, I have tested it by multiple function calls and it's not creating new iframes for every preview.

File: screenshot-iframe.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import {storageHost} from './constants';
+
+export const getScreenshotBlob = async (reqUrl, hash, ext, type, width, height) => {
```
Comment (abeersaqib) [inline screenshot-iframe.js:3]: you are assuming here that hash is the second parameter this means if the user has to pass an argument for hash 
he will using like this?

getScreenshotBlob(undefined, 'some hash') ?
it looks not good to me 

Check for the first paramter whether it's a URL or hash and based on that construct the reqUrl

File: screenshot-iframe.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import {storageHost} from './constants';
```
Comment (abeersaqib) [inline screenshot-iframe.js:1]: Change the filename to preview.js

File: screenshot-iframe.js (hunk 6 above)
Comment (abeersaqib) [inline screenshot-iframe.js:3]: Change name of function to generatePreview

**Review by abeersaqib — CHANGES_REQUESTED** [review]

File: screenshot-iframe.js (hunk 7 above)
Comment (mujabmuneeb) [reply screenshot-iframe.js]: Done

File: screenshot-iframe.js (hunk 6 above)
Comment (mujabmuneeb) [reply screenshot-iframe.js]: Done

Comment (mujabmuneeb) [reply screenshot-iframe.js]: Got it, fixed.

File: preview.js [hunk 8]
```diff
@@ -1,476 +1,39 @@
...
+    if(!isValidURL(reqUrl)) {
+        reqUrl = `${storageHost}/ipfs/${reqUrl}`;
+    }
+    var ssUrl = `${window.origin}/screenshot.html?url=${reqUrl}&ext=${ext}&type=${type}&width=${width}&height=${height}`;
+    
+    window.open(ssUrl, 'screenshotIframe')
+
+    return new Promise(function (resolve) {
+
+    var f = function(event){
```
Comment (avaer) [inline preview.js:24]: Messed up spacing.

Comment (mujabmuneeb) [reply preview.js]: Fixed.

**Pull request closed without merging**